x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/string/SubstringTests.java

```diff
@@ -111,6 +111,10 @@ public void testWholeString() {
         assertThat(process("a tiger", 1, null), equalTo("a tiger"));
     }
 
+    public void testEmptyString() {
+        assertThat(process("", 1, 3), equalTo(""));
+    }
```

Comment:
Best I think to make this a `TestCase` that's always empty.

Reply:
I supposed that `process()` was made to avoid adding testcases for each thing (?)
I've been reading a big how those suppliers work, and I feel like adding this case there would add a lot of overhead (A lot of null/wrong types checks) just to test this

Reply:
It wouldn't add any wrong type checks because is just checking against existing signatures. This wouldn't add another unique signature. It'd add a few extra null checks, though you could skip that if you add it to the list after the nulls are inserted and that's fine.

And, yeah, `process` was added before we had those test cases. We're moving everything into the test cases.